include/roboptim/core/function/polynomial.hh

```diff
@@ -72,6 +72,16 @@ namespace roboptim
     value_type applyPolynomial
     (const_vector_ref coeffs, const_argument_ref x) const;
 
+  public:
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Flag representing the Roboptim Function type
+    static const unsigned long flag_;
+
```

Comment:
Not sure about that. The functions the plugins are supposed to use are always one of GenericFunction's derived classes only, so the protected members should be accessible.